tests/test_validator.py

```diff
@@ -92,3 +95,85 @@ def test_type_alias():
+        x: int
+
+    lax_validator = AnalyzedType(Model, config=ConfigDict(strict=False))
+    strict_validator = AnalyzedType(Model, config=ConfigDict(strict=True))
+
+    assert lax_validator.validate_python({'x': '1'}, strict=None) == Model(x=1)
+    assert lax_validator.validate_python({'x': '1'}, strict=False) == Model(x=1)
+    with pytest.raises(ValidationError) as exc_info:
+        lax_validator.validate_python({'x': '1'}, strict=True)
+    assert exc_info.value.errors() == [
+        {'type': 'int_type', 'loc': ('x',), 'msg': 'Input should be a valid integer', 'input': '1'}
+    ]
+
+    with pytest.raises(ValidationError) as exc_info:
+        strict_validator.validate_python({'x': '1'})
+    assert exc_info.value.errors() == [
+        {'type': 'int_type', 'loc': ('x',), 'msg': 'Input should be a valid integer', 'input': '1'}
+    ]
+    assert strict_validator.validate_python({'x': '1'}, strict=False) == Model(x=1)
+    with pytest.raises(ValidationError) as exc_info:
+        strict_validator.validate_python({'x': '1'}, strict=True)
+    assert exc_info.value.errors() == [
+        {'type': 'int_type', 'loc': ('x',), 'msg': 'Input should be a valid integer', 'input': '1'}
+    ]
+
+
+def test_validate_json_strict() -> None:
+    class Model(TypedDict):
+        x: int
+
+    lax_validator = AnalyzedType(Model, config=ConfigDict(strict=False))
+    strict_validator = AnalyzedType(Model, config=ConfigDict(strict=True))
+
+    assert lax_validator.validate_json(json.dumps({'x': '1'}), strict=None) == Model(x=1)
+    assert lax_validator.validate_json(json.dumps({'x': '1'}), strict=False) == Model(x=1)
+    with pytest.raises(ValidationError) as exc_info:
+        lax_validator.validate_json(json.dumps({'x': '1'}), strict=True)
+    assert exc_info.value.errors() == [
+        {'type': 'int_type', 'loc': ('x',), 'msg': 'Input should be a valid integer', 'input': '1'}
+    ]
+
+    with pytest.raises(ValidationError) as exc_info:
+        strict_validator.validate_json(json.dumps({'x': '1'}), strict=None)
+    assert exc_info.value.errors() == [
+        {'type': 'int_type', 'loc': ('x',), 'msg': 'Input should be a valid integer', 'input': '1'}
+    ]
+    assert strict_validator.validate_json(json.dumps({'x': '1'}), strict=False) == Model(x=1)
+    with pytest.raises(ValidationError) as exc_info:
+        strict_validator.validate_json(json.dumps({'x': '1'}), strict=True)
+    assert exc_info.value.errors() == [
+        {'type': 'int_type', 'loc': ('x',), 'msg': 'Input should be a valid integer', 'input': '1'}
+    ]
+
+
+def test_merge_config() -> None:
+    class Model(BaseModel):
+        x: int
+        y: str
+
+        model_config = ConfigDict(strict=True, title='FooModel')
+
+    analyzed = AnalyzedType(Model, config=ConfigDict(strict=False, str_max_length=20))
+
+    # strict=False gets applied to the outer Model but not to the inner typeddict validator
+    # so we're allowed to validate a dict but `x` still must be an int
```

Comment:
This is an interesting artifact of how we validate models by wrapping a typeddict schema. I would not expect this as a user. I would expect that `model_config` has the same meaning as the `config` argument to `AnalyzedType`.